spec.emu

```diff
@@ -882,6 +957,217 @@ contributors: Google, Ecma International
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It returns an ISO Date Record that, when converted to a Calendar Date Record with _calendar_ (for example, with CalendarISOToDate), contains the given _year_, _arithmeticalMonth_, and _day_ values in its [[Year]], [[Month]], and [[Day]] fields. If this is not possible, a RangeError is thrown.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. If _year_, _arithmeticalMonth_, and _day_ do not form a valid date in _calendar_, throw a RangeError exception.
+        1. Return a calendar-dependent ISO Date Record as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-calendardaysinmonth" type="implementation-defined abstract operation">
+      <h1>
+        CalendarDaysInMonth (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _year_: an integer,
+          _arithmeticalMonth_: a positive integer,
+        ): an integer
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It interprets _year_ and _arithmeticalMonth_ as arithmetical values in the given _calendar_ and returns the number of days in the corresponding month.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Return a calendar-dependent integer as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-calendarmonthsinyear" type="implementation-defined abstract operation">
+      <h1>
+        CalendarMonthsInYear (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _year_: an integer,
+        ): an integer
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It interprets _year_ as arithmetical value in the given _calendar_ and returns the number of months in the corresponding year.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Return a calendar-dependent integer as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-balancenonisodate" type="implementation-defined abstract operation">
+      <h1>
+        BalanceNonISODate (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _year_: an integer,
+          _arithmeticalMonth_: an integer,
+          _day_: an integer,
+        ): a Record with fields [[Year]] (an integer), [[Month]] (an integer), and [[Day]] (an integer)
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It interprets the given _year_ and potentially out-of-range _arithmeticalMonth_ and _day_ as arithmetical values in the given _calendar_ and returns in-range values by overflowing out-of-range _arithmeticalMonth_ or _day_ values into the next-highest unit.
```

Comment:
```suggestion
          It interprets the given _year_ and potentially out-of-range _arithmeticalMonth_ and/or _day_ as arithmetical values in the given _calendar_ and returns in-range values by overflowing out-of-range _arithmeticalMonth_ or _day_ values into the next-highest unit.
```

There's a minor ambiguity possible where it's unclear whether the `_day_`could also be out of range. Using and/or might clarify,.